plugins/node/opentelemetry-instrumentation-mysql2/src/instrumentation.ts

```diff
@@ -149,11 +147,13 @@ export class MySQL2Instrumentation extends InstrumentationBase<MySQL2Instrumenta
               );
             }
           }
-
           span.end();
         });
 
-        if (arguments.length === 1) {
+        if (
+          arguments.length === 1 ||
+          (arguments.length === 2 && typeof arguments[1] !== 'function')
+        ) {
```

Comment:
This is an interesting fix that could be mentioned in the PR description. IIUC, before this change, the following usage would result in an unended span:

```ts
        const sql = 'SELECT 1+? as solution';
        const query = connection.query(sql, [2]);
```

Reply:
not sure it _was_ a bug, I had to add it since some existing tests were failing. Anyway, looking at it after your comment, I rewrote the statement to simplify it.